### PR TITLE
Milestone2: Create Components for Calculator

### PR DESCRIPTION
**Pull request by anewman15**
In this PR, I implemented the reqs for second milestone. They include creating
- App component that houses the app and exports itself to be imported by index
- Display component to be included in App
- ButtonPanel component that houses the Button instances created using Button component
- Button component to create Buttons for the ButtonPanel

### CODE REVIEW

File: src/components/Display.js [hunk 1]
```diff
@@ -0,0 +1,9 @@
+import React from 'react';
+
+export default class Display extends React.Component {
+  render() {
+    return (
+      <div>{this.props.calcResult}</div>
+    );
+  }
+}
```
Comment (PhillipUg) [inline src/components/Display.js]: - [x] Make this a functional component
* [x] it accepts `the result of the calculation` as an incoming prop ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/react_reducx/calculator/milestone2/display/incomingProps.png))
* [x] its result prop should be a String with default value = 0 ([should be String screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/react_reducx/calculator/milestone2/display/validateAsString.png)) ([default props screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/react_reducx/calculator/milestone2/display/DefaultProp.png))

File: src/components/ButtonPanel.js [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+      <Button btnName="+" />
+    </div>
+    <div>
+      <Button btnName="0" />
+      <Button btnName="." />
+      <Button btnName="=" />
+    </div>
+  </div>
+);
+
```
Comment (PhillipUg) [inline src/components/ButtonPanel.js]: - [x] There is a better way of doing this. Try putting all button types in an array and looping over that array to render the buttons. Thats way, you reduce on code duplication

File: src/components/Button.js [hunk 3]
```diff
@@ -0,0 +1,11 @@
...
+
+export default class Button extends React.Component {
+  render() {
+    return (
+      <button>
+        {this.props.btnName}
+      </button>
+    );
+  }
+}
```
Comment (PhillipUg) [inline src/components/Button.js]: - [x] Make this a functional component as well.

**Review by PhillipUg — CHANGES_REQUESTED** [review]
Hey team, 👋

You have done an amazing job implementing the requirements for this project.

However, there are still some changes 👇 you need to implement to approve this request 🔍

### Changes Required ♻️ 

- [ ] At this stage, `App` should be a functional component and all the other components that don't have state should be changed to functional components.

I have left a couple of inline comments below

#### Please, make all REQUIRED changes and submit your request again! 😀

If you have any questions, feel free to send me a message through Slack as **Phillip Musiime** or
 <a href="mailto:phillipmusiime@gmail.com"> gmail</a> &nbsp; &nbsp; [Twitter Follow](https://twitter.com/intent/follow?screen_name=Phillip_Ug) &nbsp; &nbsp; [GitHub followers](https://github.com/PhillipUg)

File: src/components/ButtonPanel.js (hunk 2 above)
Comment (anewman15) [reply src/components/ButtonPanel.js]: Not a good idea, because it causes issues with styling the groups of Buttons.

**Comment by anewman15** [comment]
Done

File: src/components/Button.js [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+*/
+
+import React from 'react';
+
+const Button = props => (
+  <button>
+    {props.btnName}
+  </button>
+);
+
```
Comment (mikenath223) [inline src/components/Button.js:12]: - [ ] Kindly add proptypes validation here. Its button name prop should be a String.

**Review by mikenath223 — CHANGES_REQUESTED** [review]
### Hello @anewman15 👋

**Good job here so far. You've fixed a couple of issues raised in the previous review. 👍**

However, let's make some fixes.

- Please refer to comments on the code.

### **STATUS: CHANGES REQUESTED** 👷‍♂️

Kindly fix these issues and request another review👍
Feel free to contact me on slack(michgolden) if you need further clarification.

**HAPPY CODING 🎉💻🎉**

**Michgolden Ukeje (CR)**

File: src/components/Button.js (hunk 4 above)
Comment (mikenath223) [inline src/components/Button.js]: - [ ] Please add proptypes validation here. Its button name prop should be a String.

File: src/components/Display.js [hunk 5]
```diff
@@ -0,0 +1,11 @@
...
+ eslint-disable no-unused-vars
+*/
+
+import React from 'react';
+
+const Display = props => (
+  <div>{props.calcResult}</div>
+);
+
+export default Display;
```
Comment (mikenath223) [inline src/components/Display.js]: - [ ] As stated here: https://github.com/anewman15/react-calculator/pull/3#discussion_r573563403. Please add proptypes validation here.

**Review by MuriloRoque — APPROVED** [review]
## Status: Approved :green_circle: 

### Hello @anewman15 :wave: 

### Thank you for making all of the requested changes. Your project is approved :clap: :clap: :clap: 

### You may merge, good luck with your next projects :rocket: 

Murilo Roque Paiva da Silva

- Github: [@muriloroque](https://github.com/MuriloRoque)
- Linkedin: [MuriloRoque](https://www.linkedin.com/in/murilo-roque-b1268741/)

[Twitter Follow](https://twitter.com/MuriloRoquePai1)

**Pull request merged**